### PR TITLE
feat: Sync `Pool/ResultPool` & `Wait`

### PR DESCRIPTION
**Pull request by MatsuoTakuro**

### CODE REVIEW

File: sync/pool.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+		},
+		max:              0,
+		cancelOnFirstErr: false,
+	}
+
+	for _, opt := range opts {
+		opt(plOpts)
+	}
+
+	// set setry transaction or span
```
Comment (zhongdalu-trechina) [inline sync/pool.go:74]: typo: `sentry`

File: sync/result_pool.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	if plOpts.collectErroredRlts {
+		pl = pl.WithCollectErrored()
+	}
+	if plOpts.pl.max > 0 {
+		pl.WithMaxGoroutines(plOpts.pl.max)
+	}
+
+	return ResultPool[T]{
+		pool: pl,
+		span: nil,
```
Comment (zhongdalu-trechina) [inline sync/result_pool.go:91]: should be `span`

Comment (MatsuoTakuro) [reply sync/result_pool.go]: Really thanks for reviewing!
Fixed both as you pointed out.

**Review by zhongdalu-trechina — APPROVED** [review]
LGTM

**Pull request merged**